lms/djangoapps/grades/tasks.py

```diff
@@ -40,19 +38,21 @@ def recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher, r
         - raw_possible: the max raw points the leaner could have earned
         on the problem
     """
-    if not PersistentGradesEnabledFlag.feature_enabled(course_id):
+    course_key = CourseLocator.from_string(course_id)
+    if not PersistentGradesEnabledFlag.feature_enabled(course_key):
         return
 
-    course_key = CourseLocator.from_string(course_id)
     scored_block_usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
     score = get_score(user_id, scored_block_usage_key)
 
     # If the score is None, it has not been saved at all yet
     # and we need to retry until it has been saved.
     if score is None:
-        _retry_recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher, raw_earned, raw_possible)
+        raise _retry_recalculate_subsection_grade(
+            user_id, course_id, usage_id, only_if_higher, raw_earned, raw_possible,
+        )
     else:
-        module_raw_earned, module_raw_possible = score  # pylint: disable=unpacking-non-sequence
+        module_raw_earned, module_raw_possible = score
```

Comment:
FYI @sanfordstudent I was able to get rid of the pylint warning by moving the `raise` call out of `_retry_recalculate_subsection_grade`.  I think this is a good move since it's easier for the reader to understand that the retry is being raised and the rest of the method is not getting called.  This also satisfies pylint as it is also able to parse and understand the same method code flow.